### PR TITLE
fix: Scanning '%' from cloud

### PR DESCRIPTION
**Pull request by nameexhaustion**
Fixes https://github.com/pola-rs/polars/issues/17888

### CODE REVIEW

File: crates/polars-io/src/path_utils/hugging_face.rs [hunk 1]
```diff
@@ -251,7 +252,11 @@ pub(super) async fn expand_paths_hf(
         );
         let rel_path = path_parts.path.as_str();
 
-        let (prefix, expansion) = extract_prefix_expansion(rel_path)?;
+        let (prefix, expansion) = if glob {
```
Comment (nameexhaustion) [inline crates/polars-io/src/path_utils/hugging_face.rs:255]: drive-by - `glob=False` for Hugging Face

**Review by ritchie46 — APPROVED** [review]
